pkg/interfaces/service.go

```diff
@@ -69,6 +84,11 @@ type ApplicationService interface {
 	// Note that the functions are executed in the order provided in the list.
 	// An error is returned if the list is empty.
 	SetFunctionsPipeline(transforms ...AppFunction) error
+	// AddFunctionsPipelineForTopic adds a functions pipeline with the specified unique id and list of Application Functions
+	// to be executed when the incoming topic matches the specified topic. The specified topic may contain the '#' wildcard
+	// so that it matches multiple incoming topics. If just "#" is used for the specified topic it will match all incoming
+	// topics and the specified functions pipeline will execute on every message received.
+	AddFunctionsPipelineForTopic(id string, topic string, transforms ...AppFunction) error
```

Comment:
I think this should be SetFunctionsPipelineForTopic for consistency/discoverability purposes